app/notification.go

```diff
@@ -598,6 +598,32 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 					return nil, errors.Wrapf(err, "cannot get thread %q for user %q", post.RootId, uid)
 				}
 				if userThread != nil {
+					previousUnreadMentions := userThread.UnreadMentions
+					previousUnreadReplies := max(userThread.UnreadReplies-1, 0)
+
+					if mentions.isUserMentioned(uid) {
+						previousUnreadMentions = max(userThread.UnreadMentions-1, 0)
+					}
+
+					// set last viewed at to now for commenter
+					if uid == post.UserId {
+						opts := store.ThreadMembershipOpts{
+							UpdateViewedTimestamp: true,
+						}
+						// should set unread mentions, and unread replies to 0
+						um, err := a.Srv().Store.Thread().MaintainMembership(uid, post.RootId, opts)
+
+						if err != nil {
+							return nil, errors.Wrapf(err, "cannot maintain thread membership %q for user %q", post.RootId, uid)
+						}
+
+						userThread.UnreadMentions = um.UnreadMentions                                        // should be 0
+						userThread.UnreadReplies, err = a.Srv().Store.Thread().GetThreadUnreadReplyCount(um) // should be 0
```

Comment:
If they should be 0, why are we making the store call? Can we not set to 0, and write a test to lock the behavior?

Reply:
I was thinking about that, and wasn't sure if it will be prone to error or not, I'll try to break a test to see if we are safe and then change UnreadReplies to 0 instead of the store call

Reply:
done, `TestThreadSocketEvents` cover this behavior, and will break if something changes